packages/dds/tree/src/test/feature-libraries/modular-schema/typedSchema/example.spec.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    typedTreeSchema as tree,
+    typedFieldSchema as field,
+    // Allow importing from this specific file which is being tested:
+    /* eslint-disable-next-line import/no-internal-modules */
+} from "../../../../feature-libraries/modular-schema/typedSchema";
+
+import { rootFieldKey, SchemaDataAndPolicy, ValueSchema } from "../../../../core";
+import { defaultSchemaPolicy, FieldKinds } from "../../../../feature-libraries";
+
+// Aliases for conciseness
+const { optional, value, sequence } = FieldKinds;
+
+// Declare a simple type which just holds a number.
+const numberSchema = tree({
+    name: "number",
+    value: ValueSchema.Number,
+});
+
+// Declare an aggregate type with local fields
+const ballSchema = tree({
+    name: "Ball",
+    local: {
+        x: field(value, [numberSchema]),
+        y: field(value, [numberSchema]),
+    },
+});
+
+// We can inspect the schema.
+// Note that the inferred type here actually includes the FieldKind's editor,
+// So it would be possible to derive a type safe editing API from this type.
```

Comment:
is it in a state where you could just demonstrate this with some added lines? might be worth doing so. it's an interesting observation.

Reply:
For our real use-cases it needs the TODO down below (to access field content type). Doing that and making a typed API (probably for EditableTree) is the next step.

Reply:
I went ahead and tried to make such an example. Doing so uncovered an bunch of issues, but I got it working.